zerver/tests/test_events.py

```diff
@@ -1022,17 +1022,19 @@ def test_change_realm_message_edit_settings(self):
             ('data', check_dict_only([
                 ('allow_message_editing', check_bool),
                 ('message_content_edit_limit_seconds', check_int),
+                ('edit_topics_by_admins_only', check_bool),
             ])),
         ])
         # Test every transition among the four possibilities {T,F} x {0, non-0}
-        for (allow_message_editing, message_content_edit_limit_seconds) in \
-            ((True, 0), (False, 0), (True, 0), (False, 1234), (True, 0), (True, 1234), (True, 0),
-             (False, 0), (False, 1234), (False, 0), (True, 1234), (False, 0),
-             (True, 1234), (True, 600), (False, 600), (False, 1234), (True, 600)):
+        for (allow_message_editing, message_content_edit_limit_seconds, edit_topics_by_admins_only) in \
+            ((True, 0, False), (False, 0, False), (True, 0, False), (False, 1234, False), (True, 0, False), (True, 1234, False), (True, 0, False),
+             (False, 0, False), (False, 1234, False), (False, 0, False), (True, 1234, False), (False, 0, False),
+             (True, 1234, False), (True, 600, False), (False, 600, False), (False, 1234, False), (True, 600, False)):
             events = self.do_test(
                 lambda: do_set_realm_message_editing(self.user_profile.realm,
                                                      allow_message_editing,
-                                                     message_content_edit_limit_seconds))
+                                                     message_content_edit_limit_seconds,
+                                                     edit_topics_by_admins_only))
```

Comment:
Agreed; the 2 settings already there are directly coupled in the implementation, this one isn't, so rpobably best to avoid it complicating this loop.

We probably want also a simple test for the new setting.